File: base/src/main/java/me/soushin/tinmvparms/base/app/base/BaseActivity.java
package me.soushin.tinmvparms.base.app.base;

import android.content.Context;
import android.content.Intent;
import androidx.annotation.NonNull;
import com.hjq.toast.ToastUtils;
import com.jess.arms.mvp.IPresenter;

import static com.jess.arms.utils.Preconditions.checkNotNull;
/**
 * 自定义的Activity基类
 * @author Soushin
 * @time 2019/12/26 11:23
 */
public abstract class BaseActivity<P extends IPresenter> extends com.jess.arms.base.BaseActivity<P>{

    public void showMessage(@NonNull String message) {
        checkNotNull(message);
        ToastUtils.show(message);
    }

    public void launchActivity(@NonNull Intent intent) {
        checkNotNull(intent);
        startActivity(intent);
    }

    public void showLoading(){

    }

    public void hideLoading(){

    }

    public void goActivity(Class clazz) {
        startActivity(new Intent(this, clazz));
    }

    public void killMyself() {
        finish();
    }

    public Context getThis() {
        return this;
    }

}

File: base/src/main/java/me/soushin/tinmvparms/base/app/base/BaseActivity.kt
package me.soushin.tinmvparms.base.app.base

import android.content.Context
import android.content.Intent

import com.hjq.toast.ToastUtils
import com.jess.arms.mvp.IPresenter


import com.jess.arms.utils.Preconditions.checkNotNull

/**
 * 自定义的Activity基类
 * @author Soushin
 * @time 2019/12/26 11:23
 */
abstract class BaseActivity<P : IPresenter> : com.jess.arms.base.BaseActivity<P>() {

    val `this`: Context
        get() = this

    fun showMessage(message: String) {
        checkNotNull(message)
        ToastUtils.show(message)
    }

    open fun launchActivity(intent: Intent) {
        checkNotNull(intent)
        startActivity(intent)
    }

    open fun showLoading() {

    }

    open fun hideLoading() {

    }

    fun goActivity(clazz: Class<*>) {
        startActivity(Intent(this, clazz))
    }

    open fun killMyself() {
        finish()
    }

    /**
     * 默认不使用EventBus
     */
    override fun useEventBus(): Boolean {
        return false
    }

}
